src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -502,6 +505,87 @@ public OrganisationEntityResponseV2 retrieveOrganisationForV2Api(
         return new OrganisationEntityResponseV2(organisation, true, isPendingPbaRequired, false, true);
     }
 
+    @Override
+    public MultipleOrganisationsResponse retrieveOrganisationsByProfileIds(List<String> organisationProfileIds,
+                                                                           UUID searchAfter) {
+        List<String> orgTypes = convertProfileIdsToOrgTypes(organisationProfileIds);
+
+        boolean orgIdFilterProvided = !orgTypes.isEmpty();
+        boolean searchAfterProvided = searchAfter != null;
+
+        List<Organisation> organisations = new ArrayList<>();
+
+        if (orgIdFilterProvided && searchAfterProvided) {
+            organisations = organisationRepository.findByOrgTypeInAndIdGreaterThan(orgTypes, searchAfter);
+        }
+
+        if (orgIdFilterProvided && !searchAfterProvided) {
+            organisations = organisationRepository.findByOrgTypeIn(orgTypes);
+        }
+
+        if (!orgIdFilterProvided && searchAfterProvided) {
+            organisations = organisationRepository.findByIdGreaterThan(searchAfter);
+        }
+
+        if (!orgIdFilterProvided && !searchAfterProvided) {
+            organisations = organisationRepository.findAll();
```

Comment:
is this the requirement to fetch all organisations if ordId filter and search after is not provided.

this will cause the performance issues

Reply:
resolved with a new query without joins